classifier_and_summarizer/summarizer/summarizer.py

```diff
@@ -0,0 +1,179 @@
+''' This Module provides the main interface
+for the Summarizer module
+'''
+from classifier_and_summarizer.summarizer.budgeted_max_cover.budgeted_max_cover_interface import \
+    BudgetedMaxCoverInterface
+from classifier_and_summarizer.summarizer.extractor_output_preprocessor import \
+    SentenceWithAttributes
+from classifier_and_summarizer.summarizer.summarizer_output import (StampPage,
+                                                                    StampPages)
+from classifier_and_summarizer.summarizer.text_media_matching.text_media_matching_interface import \
+    TextMediaMatchingInterface
+from data_models.contents import ContentType
+
+
+class Summarizer:
+    def __init__(
+            self,
+            title_text_contents,
+            normal_text_contents,
+            media_contents,
+            embedded_contents,
+            max_pages_allowed):
+        self.title_text_contents = title_text_contents
+        self.normal_text_contents = normal_text_contents
+        self.media_contents = media_contents
+        self.embedded_contents = embedded_contents
+        self.max_pages_allowed = max_pages_allowed
+        # we don't directly instantiate StampPages
+        # object since we need to use the list of
+        # stamp pages to cap and pick stamp pages
+        self.stamp_pages_list = list()
+        self.stamp_pages = StampPages()
+
+        # collect summary sentence embeddings
+        self.summary_sentence_embeddings = [
+            text.embedding for text in self.normal_text_contents
+        ]
+
+    def get_summarized_content(self):
+        # first do text media matching
+        self._perform_text_media_matching()
+
+        # concat the matched and unmatched contents list
+        # and send to make stamp page objects out of them
+        self._assemble_and_add_stamp_pages_to_list(
+            self.matched_contents + self.unmatched_contents
```

Comment:
Where are self.matched_contents and self.unmatched_contents initialized?

Reply:
defined in `_perform_text_media_matching( )` method